engine/src/test/java/org/hibernate/search/testsupport/TestConstants.java

```diff
@@ -89,4 +85,11 @@ public static String getIndexDirectory(Class<?> testClass) {
 	public static boolean arePerformanceTestsEnabled() {
 		return Boolean.getBoolean( "org.hibernate.search.enable_performance_tests" );
 	}
+
+	private static class CallerProvider extends SecurityManager {
+
+		public Class<?> getCallerClass() {
+			return getClassContext()[2];
```

Comment:
oops, sorry got confused by the diff. All clear now :)


On 13 November 2015 at 18:07, Gunnar Morling <notifications@github.com>
wrote:

> In
> engine/src/test/java/org/hibernate/search/testsupport/TestConstants.java
> <https://github.com/hibernate/hibernate-search/pull/946#discussion_r44813594>
> :
>
> > @@ -89,4 +85,11 @@ public static String getIndexDirectory(Class<?> testClass) {
> >  	public static boolean arePerformanceTestsEnabled() {
> >  		return Boolean.getBoolean( "org.hibernate.search.enable_performance_tests" );
> >  	}
> > +
> > +	private static class CallerProvider extends SecurityManager {
> > +
> > +		public Class<?> getCallerClass() {
> > +			return getClassContext()[2];
>
> But CallerProvider is stateless, it does not point to any directory. It
> is just used to obtain the method of the calling test which wants to create
> an index directory (so its name can be added to the dir name). It's not
> related to the target dir.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hibernate/hibernate-search/pull/946/files#r44813594>.
>